cmd/talosctl/cmd/talos/root.go

```diff
@@ -123,3 +129,103 @@ var Commands []*cobra.Command
 func addCommand(cmd *cobra.Command) {
 	Commands = append(Commands, cmd)
 }
+
+// completeResource represents tab complete options for `ls` and `ls *` commands.
+func completePathFromNode(inputPath string) []string {
+	pathToSearch := inputPath
+
+	// If the pathToSearch is empty, use root '/'
+	if pathToSearch == "" {
+		pathToSearch = "/"
+	}
+
+	var paths map[string]struct{}
+
+	// search up one level to find possible completions
+	if pathToSearch != "/" && !strings.HasSuffix(pathToSearch, "/") {
+		index := strings.LastIndex(pathToSearch, "/")
+		// we need a trailing slash to search for items in a directory
+		pathToSearch = pathToSearch[:index] + "/"
+	}
+
+	paths = getPathFromNode(pathToSearch, inputPath)
+
+	result := make([]string, 0, len(paths))
+
+	for k := range paths {
+		if strings.HasPrefix(k, inputPath) {
```

Comment:
do we need this check if `getPathFromNode` already filtered the output?

Reply:
not needed, forgot about it here, fixed